diff --git a/src/backends/meta-backend.c b/src/backends/meta-backend.c
index 8bb5e0e..2c1d2ce 100644
--- a/src/backends/meta-backend.c
+++ b/src/backends/meta-backend.c
@@ -1104,8 +1104,7 @@ update_pointer_visibility_from_event (MetaBackend  *backend,
     case CLUTTER_PEN_DEVICE:
     case CLUTTER_ERASER_DEVICE:
     case CLUTTER_CURSOR_DEVICE:
-      if (time_ms > priv->last_pointer_motion + HIDDEN_POINTER_TIMEOUT)
-        set_cursor_visible (backend, FALSE);
+      set_cursor_visible (backend, FALSE);
       break;
     case CLUTTER_KEYBOARD_DEVICE:
     case CLUTTER_PAD_DEVICE:
diff --git a/src/backends/meta-sprite.c b/src/backends/meta-sprite.c
index d8264ca..ca4f0b3 100644
--- a/src/backends/meta-sprite.c
+++ b/src/backends/meta-sprite.c
@@ -117,6 +117,10 @@ meta_sprite_update_cursor (ClutterSprite *sprite,
                                        pointer_visible ? cursor : NULL);
       meta_cursor_tracker_set_current_cursor (cursor_tracker, cursor);
     }
+  else if (clutter_sprite_get_role (sprite) == CLUTTER_SPRITE_ROLE_TABLET)
+    {
+      meta_cursor_renderer_set_cursor (cursor_renderer, NULL);
+    }
   else
     {
       meta_cursor_renderer_set_cursor (cursor_renderer, cursor);
